Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -122,6 +122,95 @@ open class MongoSwiftTestCase: XCTestCase {
     }
 }
 
+/// Enumerates the different topology configurations that are used throughout the tests
+public enum TestTopologyConfiguration {
+    case sharded
+    case replicaSet
+    case single
+    case unknown
+
+    public init(from str: String) {
+        switch str {
+        case "sharded", "sharded_cluster":
+            self = .sharded
+        case "replicaset", "replica_set", "replicaSet":
+            self = .replicaSet
+        case "single":
+            self = .single
+        default:
+            self = .unknown
+        }
+    }
+}
+
+/// Enumerates different possible unmet requirements that can be returned by meetsRequirements
+public enum UnmetRequirements {
+    case minServerVersion(actual: ServerVersion, required: ServerVersion)
+    case maxServerVersion(actual: ServerVersion, required: ServerVersion)
+    case topology(actual: TestTopologyConfiguration, required: [TestTopologyConfiguration])
+}
+
+/// Determines the topologyType of a client based on the reply returned by running an isMaster command
+public func topologyType(_ isMasterReply: BSONDocument) throws -> TestTopologyConfiguration {
```

Comment:
rather than a free floating function, this would be a little more discoverable as an initializer on `TestTopologyConfiguration`.

Reply:
done